### PR TITLE
Profile Picture and Display Name backend integration for Onboarding Landing Page

### PR DESCRIPTION
**Pull request by wimaan3**

### CODE REVIEW

File: packages/UI/src/context/UserDataContext.js [hunk 1]
```diff
@@ -32,11 +32,11 @@ function useUserDataHook() {
   }, [userData.id > 0]);
 
   useEffect(() => {
-    axios(`${process.env.NEXT_PUBLIC_STRAPI_URL}/users/me?populate=deep`, {
+    axios(`${process.env.NEXT_PUBLIC_STRAPI_URL}/users/80?populate=deep`, {
```
Comment (Enjoy2Live) [inline packages/UI/src/context/UserDataContext.js:35]: please revert this, we should have this call /users/me not /users/80

Comment (wimaan3) [reply packages/UI/src/context/UserDataContext.js]: Fixed it, thanks for catching that.

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js [hunk 2]
```diff
@@ -1,272 +1,385 @@
...
+  }
 
-    function validate(formValue) {
-        return {
-            isFirstNameValid: formValue.firstNameTouched && /^[a-z ,.'-]+$/i.test(formValue.firstName),
-            isLastNameValid: formValue.lastNameTouched && /^[a-z ,.'-]+$/i.test(formValue.lastName),
-            isFormValid: function () { return formValue.termsAndConditions && this.isFirstNameValid && this.isLastNameValid }
+  function handleFileChange(event) {
+    const file = event.target.files[0];
+    if (file) {
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:83]: Could we combine this and the next if check using `if (file?.type === 'image/png' || file?.type === 'image/jpeg') `?

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js [hunk 3]
```diff
@@ -1,272 +1,385 @@
...
     }
+  }
 
-    function onFirstNameChange(e) {
-        console.log(person);
-        setPerson({ ...person, firstName: e.target.value, firstNameTouched: true });
-        console.log("VALIDATION: ", formValidation)
-    }
+  function onFirstNameChange(e) {
+    console.log(person);
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:101]: Will the debug logs be removed later?

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js [hunk 4]
```diff
@@ -1,272 +1,385 @@
...
 
-    const onCancelClick = (e) => {
-        router.push({
-            pathname: '/',
-        }, '/');
-        e.preventDefault()
-    }
+    const requestBody = {
+      data: {
+        displayName: `${person.firstName} ${person.lastName}`,
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:140]: Would user be able to change this later?

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js [hunk 5]
```diff
@@ -1,272 +1,385 @@
...
+        profilePictureUrl: uploadedProfilePicture || profilePicture,
+      },
+    };
+    profileId &&
+      axios
+        .put(
+          `${process.env.NEXT_PUBLIC_STRAPI_URL}/profiles/${profileId}/`,
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:147]: @Enjoy2Live I think we have a library to send requests to the backend?

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js (hunk 4 above)
Comment (wimaan3) [reply apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:140]: I am not sure if there are other ways to change the first and last name other than through the onboarding landing page but yes.

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js (hunk 5 above)
Comment (Enjoy2Live) [reply apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:147]: yes, please use the agent as follows

```
import agent from "@devlaunchers/utility/agent"

agent.profiles.put(profileId, requestBody).then(etc....)
```

keep in mind that agent is exported from `packages/utility/agent.ts` so if the object "profiles" can't be found (seems to be the case rn) you need to create it in that file and export it so you can access `agent.profiles` from within `apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js`

File: packages/UI/src/context/UserDataContext.js [hunk 6]
```diff
@@ -53,8 +53,8 @@ function useUserDataHook() {
         });
       })
       .catch((e) => {
-        console.error('failed to fetch', e);
-        setIsAuthenticated(false);
+        console.error('Failed to fetch', e);
+        setIsAuthenticated(true);
```
Comment (chungthuang) [inline packages/UI/src/context/UserDataContext.js:57]: Why do we set authenticated to true when there is an error?

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js [hunk 7]
```diff
@@ -1,272 +1,385 @@
...
-    //             <FormFieldMargin>
-    //                 <FormField
-    //                     error=""
-    //                     label="FIRST NAME"
-    //                     value={person.firstName}
-    //                     onChange={handleFirstNameChange}
-    //                     placeholder="Placeholder"
-    //                 />
-    //             </FormFieldMargin>
+  // return (
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:307]: Why is this commented out?

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js [hunk 8]
```diff
@@ -1,272 +1,385 @@
...
-                            <InputField
-                                error="Please fill in your Last Name"
-                                label="Last Name"
-                                name="Last Name"
-                                onChange={onLastNameChange}
-                                placeholder="Doe"
-                                touched={person.lastNameTouched ? !formValidation.isLastNameValid : false}
-                                required
-                            />
+  const onCancelClick = (e) => {
```
Comment (chungthuang) [inline apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:167]: For my understanding, this redirects the user to the home page?

**Comment by judeamos** [comment]
@wimaan3 I would close this PR and recreate it as a Draft, you do not need to push the userDataContext.js file 
image

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js (hunk 7 above)
Comment (wimaan3) [reply apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:307]: It was there before I started working on this file.

File: apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js (hunk 8 above)
Comment (wimaan3) [reply apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js:167]: Yes

File: packages/UI/src/context/UserDataContext.js (hunk 6 above)
Comment (wimaan3) [reply packages/UI/src/context/UserDataContext.js:57]: Not sure, I did not add this.

**Pull request closed without merging**